lib/new_relic/agent/custom_event_aggregator.rb

```diff
@@ -70,15 +72,20 @@ def create_custom_event_attributes(attributes)
             key = key[0, MAX_NAME_SIZE]
           end
 
-          # value is limited to 4095
+          # value is limited to 4095 expect for LLM content-related events
           if val.is_a?(String) && val.length > MAX_ATTRIBUTE_SIZE
-            val = val[0, MAX_ATTRIBUTE_SIZE]
+            val = val[0, MAX_ATTRIBUTE_SIZE] unless llm_exempt_event_attribute?(type, key)
```

Comment:
I think we should defer LLM specific logic to an LLM class. Then for performance reasons, I think we should also short-circuit the logic when LLM instrumentation isn't even active.

So this line becomes something like:

```ruby
val = val[0, MAX_ATTRIBUTE_SIZE] unless BaseLLMClass.full_length_attribute?(type, key)
```

And then over in your LLM base class if where you'd have what you are currently calling `llm_exempt_event_attribute?`:

```ruby
module NewRelic
  module Agent
    class LLMBaseClass
      def full_length_attribute?(type, key)
        return unless llm_instrumentation_enabled?

        # ...
      end
    end
  end
end